Jenkins.io - Create platform information section (#6685)
jenkins-infra/jenkins.io: content/doc/book/platform-information/upgrade-java-to-17.adoc

---
layout: subsection
title: Upgrading to Java 17
---

> There are a few details and steps to upgrade the JVM used to run Jenkins, more specifically from Java 11 to Java 17.
> 
> If you're upgrading the JVM used to run Jenkins, particularly from Java 11 to Java 17, there are some details you should know and precautions you should take.

Reviewer: ```suggestion
When upgrading from JVM used to run Jenkins Java 11 to Java 17, there are some details you should know and precautions you should take.
```

.Upgrading Jenkins Java Version From 11 to 17 
video::ZabUz6sl-8I[youtube,width=800,height=420]

== Backing up Jenkins

> As with any upgrade, link:/doc/book/system-administration/backing-up/#jenkins_home[we recommend backing up] `JENKINS_HOME` and testing the upgrade with the backup before performing the upgrade on your production instance.

Reviewer: As with any upgrade, we recommend:
 
. Backing up `JENKINS_HOME`.
. Testing the upgrade with your backup.
. Only after all required tests pass, performing the upgrade on your production instance.

Reviewer: I would include the link to backing up `JENKINS_HOME` in the steps below.

Reviewer: I would suggest using a numbered list to make the recommended steps very clear.

== Upgrading Jenkins

> If you need to upgrade Jenkins as well as the JVM, we recommend that you:
> 
> . Update all of your installed plugins.
> . Back up `JENKINS_HOME`.
> . Stop the Jenkins instance.
> . Upgrade the JVM on which Jenkins is running.
> ** Use a package manager to install the new JVM.
> ** Make sure the default JVM is the newly installed version.
> If it is not, run `systemctl edit jenkins` and set either the `JAVA_HOME` environment variable or the `JENKINS_JAVA_CMD` environment variable.
> . Upgrade Jenkins to the most recent version
> ** How you upgrade Jenkins is dependent upon your original Jenkins installation method.
> ** We recommend using the package manager of your system, such as `apt` or `yum`.
> . Validate the upgrade to confirm that all plugins and jobs are loaded.
> . Upgrade the required plugins.
> Refer to <<Upgrading Plugins>> for more information.

Reviewer: ```suggestion
If you need to upgrade Jenkins as well as the JVM, we recommend you:

. Update all of your installed plugins.
. Back up `JENKINS_HOME`.
. Stop the Jenkins instance.
. Upgrade the JVM on which Jenkins is running.
** Use a package manager to install the new JVM.
** Ensure the default JVM is the newly installed version.
*** If it is not, run `systemctl edit jenkins` and set either the `JAVA_HOME` environment variable or the `JENKINS_JAVA_CMD` environment variable.
. Upgrade Jenkins to the most recent version
** How you upgrade Jenkins is dependent upon your original Jenkins installation method.
+
TIP: We recommend using the package manager of your system, such as `apt` or `yum`.
. Validate the upgrade to confirm that all plugins and jobs are loaded.
. Upgrade the required plugins.
Refer to <<Upgrading Plugins>> for more information.
```

== Upgrading Plugins

> It's important not just to upgrade Jenkins and the JVM, but also to upgrade the plugins which support Java 17.
> Plugin upgrades assure compatibility with the most recent Jenkins releases.

Reviewer: ```suggestion
When upgrading the Java version for Jenkins and the JVM, it is important to upgrade all plugins that support Java 11. Plugin upgrades assure compatibility with the most recent Jenkins releases.
```

NOTE: If you discover a previously unreported issue, please let us know.
Refer to link:/participate/report-issue/#issue-reporting[our issue reporting documentation] for guidance.

== JVM version on agents

All agents must be running on the same JVM version as the controller, due to how controllers and agents communicate.

> If you're upgrading your Jenkins controller to run on Java 17, you also need to upgrade the JVM on your agents.

Reviewer: ```suggestion
If you're upgrading your Jenkins controller to run on Java 17, you must upgrade the JVM on your agents.
```
